planner/cascades/testdata/integration_suite_out.json

```diff
@@ -393,7 +393,7 @@
         "Plan": [
           "HashAgg_18 1.00 root  funcs:sum(Column#4)->Column#3",
           "└─TableReader_19 1.00 root  data:HashAgg_20",
-          "  └─HashAgg_20 1.00 cop[tikv]  funcs:sum(cast(test.t.b, decimal(10,0) BINARY))->Column#4",
+          "  └─HashAgg_20 1.00 cop[tikv]  funcs:sum(cast(test.t.b, decimal(11,1) BINARY))->Column#4",
```

Comment:
Why this is affected?

Reply:
Because the test sql is `select sum(case when a <= 0 or a > 1000 then 0.0 else b end) from t`, case when may be eliminated here.